enterprise/coderd/provisionerkeys.go

```diff
@@ -54,6 +55,20 @@ func (api *API) postProvisionerKey(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	reserved := []string{"built-in", "psk", "user-auth"}
```

Comment:
Do you think there is a place we can make this constants? I saw this also used in `dbmem` and some unit tests.

Not a hugel deal if it feels like a pain.